modules/azure/aks/defender.tf

```diff
@@ -0,0 +1,117 @@
+resource "azurerm_security_center_auto_provisioning" "this" {
+  auto_provision = "Off"
+}
+
+resource "azurerm_security_center_subscription_pricing" "containers" {
+  count         = var.defender_enabled ? 1 : 0
```

Comment:
Use for each instead of counts.

Reply:
What is the advantage?

Reply:
This is one of the few cases where count could be accepted, but we have been avoiding it for a long time now. So its just a design choice.

Reply:
OK, I think it looks weird to use a for-each on something that is on or off, but sure, I can fix that.